### PR TITLE
Feature: Navigation after DAO creation APP-1040

### PR DESCRIPTION
**Pull request by RakeshUP**
## Description

Navigation to the dashboard to a new empty state upon DAO Creation. This empty state constitutes that the transaction went through, but the subgraph has not yet picked up the data.

Task: [APP-1040](https://aragonassociation.atlassian.net/browse/APP-1040)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

### CODE REVIEW

**Review by jamesej — APPROVED** [review]
All looks good, thanks Rakesh

File: packages/web-app/src/hooks/useDaoDetails.tsx [hunk 1]
```diff
@@ -1,29 +1,44 @@
...
+export function useDaoDetails(
+  daoId: string
+): HookData<DaoDetails | undefined> & {waitingForSubgraph: boolean} {
   const {client} = useClient();
 
   const [data, setData] = useState<DaoDetails>();
   const [error, setError] = useState<Error>();
   const [isLoading, setIsLoading] = useState(false);
+  const [waitingForSubgraph, setWaitingForSubgraph] = useState(false);
+  const {network} = useWallet();
```
Comment (WissenIstNacht) [inline packages/web-app/src/hooks/useDaoDetails.tsx]: `useNetwork()`. At this point, the network must agree with the URL. This is not necessarily the case for the one from the wallet.

File: packages/web-app/src/pages/dashboard.tsx [hunk 2]
```diff
@@ -35,10 +70,119 @@ const Dashboard: React.FC = () => {
...
+      ),
+      [DaoCreationState.DAO_READY]: <IconCheckmark />,
+      [DaoCreationState.OPEN_DAO]: undefined,
+    };
+
+    if (
+      !pollForDaoData &&
+      daoCreationState === DaoCreationState.ASSEMBLING_DAO
+    ) {
+      pollForDaoData = window.setInterval(async () => {
```
Comment (WissenIstNacht) [inline packages/web-app/src/pages/dashboard.tsx]: Side-effects should in a `useEffect` (I mean, you even already have one to to deactivate the interval 😅 )

File: packages/web-app/src/context/apolloClient.tsx [hunk 3]
```diff
@@ -149,11 +155,22 @@ const pendingProposals = JSON.parse(
 );
 const pendingProposalsVar = makeVar<PendingProposals>(pendingProposals);
 
+type PendingDaoCreation = {
```
Comment (WissenIstNacht) [inline packages/web-app/src/context/apolloClient.tsx:158]: Small favour: Could you comment what the keys are? I suppose the outer is the DAO id? What's the inner, then?

File: packages/web-app/src/context/createDao.tsx [hunk 4]
```diff
@@ -52,7 +55,7 @@ const CreateDaoContext = createContext<CreateDaoContextType | null>(null);
 const CreateDaoProvider: React.FC<Props> = ({children}) => {
   const {open} = useGlobalModalContext();
   const navigate = useNavigate();
-  const {isOnWrongNetwork, provider} = useWallet();
+  const {isOnWrongNetwork, provider, network} = useWallet();
```
Comment (WissenIstNacht) [inline packages/web-app/src/context/createDao.tsx]: I just wrote a huge comment about this and my IDE crashed. Comment gone. Sorry, you'll have to be ok with a TLDR 😅

This should go via `useNetwork`. It should NOT go through the wallet's network. (Reason: Wallet network mayb be set independetly of what is selected on the UI.).

If you're ok with that I can push a quick commit with the necessary changes.

**Review by WissenIstNacht — CHANGES_REQUESTED** [review]
Hey thanks a lot for the PR. Unfortunately there's a couple of issues related to the handling of the network.

File: packages/web-app/src/context/apolloClient.tsx (hunk 3 above)
Comment (RakeshUP) [reply packages/web-app/src/context/apolloClient.tsx:158]: The outer is network name and the inner one is dao id

**Review by WissenIstNacht — APPROVED** [review]
Hey Rakesh, thanks a lot for the changes and clarifications.

I've made two little adjustments, hope you don't mind 🤗

**Pull request merged**